libtiledbsoma/src/soma/soma_array.cc

```diff
@@ -196,18 +196,23 @@ void SOMAArray::reset(
         mq_->select_columns(column_names);
     }
 
-    batch_size_ = batch_size;
-
-    if (result_order !=
-        ResultOrder::automatic) {  // default ResultOrder::automatic is set in
-                                   // soma_array.h
-        tiledb_layout_t layout = (result_order == ResultOrder::rowmajor) ?
-                                     TILEDB_ROW_MAJOR :
-                                     TILEDB_COL_MAJOR;
-        mq_->set_layout(layout);
-        result_order_ = result_order;
+    switch (result_order) {
+        case ResultOrder::automatic:
+            // default ResultOrder::automatic is set in soma_array.h
```

Comment:
Is there any downside to explicitly setting the value here, rather than depending on defaulting?  Also, in the case where a user is re-using the reader, and is changing the result order from "col-major" to "automatic", won't this leave the value unchanged (i.e, be a noop)?